### PR TITLE
band unit tests

### PR DESCRIPTION
**Pull request by ColinEberhardt**
Adds unit tests for the band series ... which uncovered a few bugs!

 + literal property values were not being functored
 + context and index were both incorrect

Regarding the second point, any component that uses our svg components are likely to be incorrect too.

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -236,9 +236,6 @@ module.exports = function (grunt) {
             },
             test: {
                 src: ['dist/*.js'],
-            },
```
Comment (ColinEberhardt) [inline Gruntfile.js]: I've dropped the testDebug target, now that the spec runner file is ignore, I don't see any harm in keeping this file hanging around. I find it useful for debugging.

File: src/annotation/band.js [hunk 2]
```diff
@@ -55,7 +55,11 @@
                     });
 
                 g.select('path')
```
Comment (chrisprice) [inline src/annotation/band.js]: This problem is rife throughout our use of the `fc.svg.* ` components e.g. [bar series](https://github.com/ScottLogic/d3fc/blob/master/src/series/bar.js#L45). I do think your changes represent the pattern we should be following though, should we create an issue to fix the rest?

**Comment by chrisprice** [comment]
Happy to see this merged but holding off so that you see my question.

File: src/annotation/band.js (hunk 2 above)
Comment (ColinEberhardt) [inline src/annotation/band.js]: Yep, I'll raise an issue.

**Comment by ColinEberhardt** [comment]
Merged and issue raised.

**Pull request merged**